src/Sharpliner/AzureDevOps/TemplateDefinition.cs

```diff
@@ -257,3 +260,79 @@ internal TemplateDefinition()
     {
     }
 }
+
+public abstract class TemplateDefinition<T, TParameters> : TemplateDefinition<T> where TParameters : class, new()
+{
+    protected TParameters TypedParameters { get; private set; }
```

Comment:
Do we need this?

Reply:
We need this to be able to do 2 things
- `List<Parameter> Parameters => ToParameters(TypedParameters)`
- `public static implicit operator Template<T>(TemplateDefinition<T, TParameters> definition)`

Reply:
cleaned up